src/task.c

```diff
@@ -600,9 +600,20 @@ DLLEXPORT size_t rec_backtrace_ctx(ptrint_t *data, size_t maxsize, CONTEXT *Cont
         DWORD64 ImageBase = JuliaGetModuleBase64(GetCurrentProcess(), Context->Rip);
         if (!ImageBase)
             break;
+
+        MEMORY_BASIC_INFORMATION mInfo;
+
         PRUNTIME_FUNCTION FunctionEntry = (PRUNTIME_FUNCTION)JuliaFunctionTableAccess64(GetCurrentProcess(), Context->Rip);
         if (!FunctionEntry) { // assume this is a NO_FPO RBP-based function
             Context->Rsp = Context->Rbp;                 // MOV RSP, RBP
+
+            // Check whether the pointer is valid and executable before dereferencing
+            // to avoid segfault while recording. See #10638.
+            if (VirtualQuery((LPCVOID)Context->Rsp, &mInfo, sizeof(MEMORY_BASIC_INFORMATION)) == 0)
+                break;
+            if (mInfo.AllocationProtect < PAGE_EXECUTE)
```

Comment:
> The access granted to all pages is the same (PAGE_READONLY, PAGE_READWRITE, PAGE_NOACCESS, PAGE_WRITECOPY, PAGE_EXECUTE, PAGE_EXECUTE_READ, PAGE_EXECUTE_READWRITE, PAGE_EXECUTE_WRITECOPY, PAGE_GUARD, or PAGE_NOCACHE).
https://msdn.microsoft.com/en-us/library/windows/desktop/aa366786(v=vs.85).aspx

I think this should be `X = mInfo.AllocationProtect; ((X&PAGE_READONLY) || (X&PAGE_READWRITE) || (X&PAGE_WRITECOPY) || (X&PAGE_EXECUTE_READ)) && !((X&PAGE_GUARD) || (X&PAGE_NOACCESS))`